mm2src/coins/lightning/ln_platform.rs

```diff
@@ -0,0 +1,595 @@
+            },
+        };
+        channel_manager.best_block_updated(&new_best_header, new_best_height);
+        chain_monitor.best_block_updated(&new_best_header, new_best_height);
+    }
+}
+
+pub async fn ln_best_block_update_loop(
+    platform: Arc<Platform>,
+    chain_monitor: Arc<ChainMonitor>,
+    channel_manager: Arc<ChannelManager>,
+    best_header_listener: ElectrumClient,
+    best_block: RpcBestBlock,
+) {
+    let mut current_best_block = best_block;
+    loop {
+        let best_header = match get_best_header(&best_header_listener).await {
+            Ok(h) => h,
+            Err(e) => {
+                log::error!("Error while requesting best header for lightning node: {}", e);
+                Timer::sleep(CHECK_FOR_NEW_BEST_BLOCK_INTERVAL as f64).await;
+                continue;
+            },
+        };
+        if current_best_block != best_header.clone().into() {
+            platform
+                .process_txs_unconfirmations(chain_monitor.clone(), channel_manager.clone())
+                .await;
+            platform
+                .process_txs_confirmations(
+                    best_header_listener.clone(),
+                    chain_monitor.clone(),
+                    channel_manager.clone(),
+                    best_header.block_height(),
+                )
+                .await;
+            current_best_block = best_header.clone().into();
+            update_best_block(chain_monitor.clone(), channel_manager.clone(), best_header).await;
+        }
+        Timer::sleep(CHECK_FOR_NEW_BEST_BLOCK_INTERVAL as f64).await;
+    }
+}
+
+struct ConfirmedTransactionInfo {
+    txid: Txid,
+    header: BlockHeader,
+    index: usize,
+    transaction: Transaction,
+    height: u32,
+}
+
+impl ConfirmedTransactionInfo {
+    fn new(txid: Txid, header: BlockHeader, index: usize, transaction: Transaction, height: u32) -> Self {
+        ConfirmedTransactionInfo {
+            txid,
+            header,
+            index,
+            transaction,
+            height,
+        }
+    }
+}
+
+pub struct Platform {
+    pub coin: UtxoStandardCoin,
+    /// Main/testnet/signet/regtest Needed for lightning node to know which network to connect to
+    pub network: BlockchainNetwork,
+    // Default fees to and confirmation targets to be used for FeeEstimator. Default fees are used when the call for
+    // estimate_fee_sat fails.
+    pub default_fees_and_confirmations: PlatformCoinConfirmations,
+    // This cache stores the transactions that the LN node has interest in.
+    pub registered_txs: PaMutex<HashMap<Txid, HashSet<Script>>>,
+    // This cache stores the outputs that the LN node has interest in.
+    pub registered_outputs: PaMutex<Vec<WatchedOutput>>,
+    // This cache stores transactions to be broadcasted once the other node accepts the channel
+    pub unsigned_funding_txs: PaMutex<HashMap<u64, TransactionInputSigner>>,
+}
+
+impl Platform {
+    pub fn new(
```

Comment:
I think it's excessive to add `#[inline]` for constructor functions like this one.

Reply:
I usually prefet to inline single process functions. If function executes only one process, then I prefer inlining it instead of calling the function. Since you still execute single process when you inline that function, instruction cache won't increase either. LLVM usually does this for you, but I don't like leaving it to chance.